core/src/main/java/org/jboss/jandex/IndexReaderV2.java

```diff
@@ -169,7 +170,14 @@ private void readStringTable(PackedDataInputStream stream) throws IOException {
         int size = stream.readPackedU32() + 1;
         String[] stringTable = this.stringTable = new String[size];
         for (int i = 1; i < size; i++) {
-            stringTable[i] = stream.readUTF();
+            if (version >= 11) {
+                int len = stream.readPackedU32();
+                byte[] bytes = new byte[len];
+                stream.readFully(bytes, 0, len);
+                stringTable[i] = new String(bytes, StandardCharsets.UTF_8);
```

Comment:
An optimization here could be to use InputStreamReader/OutputStreamWriter, so we avoid the large temporary array.

Reply:
Good point, I'll check that.

Reply:
Looking at it now and it actually wouldn't help. `OutputStreamWriter` writes through a `StreamEncoder`, which as the first thing copies the entire string to a `char[]`. Similarly, an `InputStreamReader` accepts a `char[]` which we'd have to provide, and constructing a `String` from it performs a copy.

So in the end, the code would be a little shorter, but would allocate the same temporary array _and then some_.